cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt

```diff
@@ -257,6 +259,43 @@ open class CropImageActivity :
         cropImageView?.rotateImage(degrees)
     }
 
+    open fun setProperCropShape(item: MenuItem) {
+        when {
+            cropImageView?.cropShape === CropImageView.CropShape.RECTANGLE -> {
+                item.icon = ResourcesCompat.getDrawable(
+                    resources, R.drawable.ic_crop_shape_circle_24,
+                    theme
+                )
+                cropImageView?.cropShape = CropImageView.CropShape.OVAL
+                cropImageView?.setFixedAspectRatio(true)
+            }
+            cropImageView?.cropShape === CropImageView.CropShape.OVAL -> {
+                item.icon = ResourcesCompat.getDrawable(
+                    resources, R.drawable.ic_crop_shape_rectangle_vertical_24,
+                    theme
+                )
+                cropImageView?.cropShape = CropImageView.CropShape.RECTANGLE_VERTICAL_ONLY
+                cropImageView?.setFixedAspectRatio(false)
+            }
+            cropImageView?.cropShape === CropImageView.CropShape.RECTANGLE_VERTICAL_ONLY -> {
+                item.icon = ResourcesCompat.getDrawable(
+                    resources, R.drawable.ic_crop_shape_rectangle_horizontal_24,
+                    theme
+                )
+                cropImageView?.cropShape = CropImageView.CropShape.RECTANGLE_HORIZONTAL_ONLY
+                cropImageView?.setFixedAspectRatio(false)
+            }
+            else -> {
+                item.icon = ResourcesCompat.getDrawable(
+                    resources, R.drawable.ic_crop_shape_square_24,
+                    theme
+                )
+                cropImageView?.cropShape = CropImageView.CropShape.RECTANGLE
+                cropImageView?.setFixedAspectRatio(true)
```

Comment:
This remove the feature of making a crop changing Vertical and Horizontal with free ratio.

Before: **Free ratio**

https://user-images.githubusercontent.com/15942449/128321118-2e42dc2a-2347-481c-bb0a-85abe0547690.mp4

After we selected square once we loose this.